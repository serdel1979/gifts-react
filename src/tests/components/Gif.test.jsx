import { screen, configure } from '@testing-library/react'
const {render} = require("@testing-library/react");
const {default: Gif} = require("../../components/Gif");

describe('Pruebas en <Gif />', () => {
  
    const title = 'Gif loco';
    const url = 'https://undom.com/';

    test('Debe hacer match con el snapshot',()=>{

        const { container } = render( <Gif title = {title} url={url}/>);
        expect( container ).toMatchSnapshot();
        
    })


    test('Debe mostrar la imagen con el URL y el ALT indicado',()=>{

        const { container } = render( <Gif title = {title} url={url}/>);
        // screen.debug();
        // expect( screen.getByRole('img').src).toBe(url);
        // expect( screen.getByRole('img').alt).toBe(title);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    })

    test('Debe mostrar el tÃ­tulo',()=>{

        const { container } = render( <Gif title = {title} url={url}/>);
        
        expect(screen.getByText(title)).toBeTruthy
    })
})
