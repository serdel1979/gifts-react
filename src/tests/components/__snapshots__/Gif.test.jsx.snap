// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Gif /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Gif loco"
      src="https://undom.com/"
    />
    <p>
      Gif loco
    </p>
  </div>
</div>
`;
